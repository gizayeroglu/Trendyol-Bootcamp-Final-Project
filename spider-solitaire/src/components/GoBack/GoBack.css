:root {
    --green:#016B02;
    --white:#E8E8E6;
}
.go-back-button {
    width: 70px;
    height: 70px;
    border-radius: 50px;
    border: none;
    outline: none;
    cursor: pointer;
    color:var(--green);
    font-weight: 700;
    top: 25px;
    left: 30px;
    position: absolute;
}
.go-back-button:hover {
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.5);
}
.go-back-icon {
    width: 50px;
    top: 10px;
    right: 10px;
    position: absolute;
}