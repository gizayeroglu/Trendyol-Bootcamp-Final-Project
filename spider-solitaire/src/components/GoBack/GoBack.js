import './GoBack.css';
import React from 'react';
import { useHistory } from 'react-router-dom';
import goBackIcon from '../../assets/go-back-icon.jpg';

function GoBack(){
  const history = useHistory();

  const handleGoHome = () => {
    history.push('/');
  };

    return(
      <button onClick={handleGoHome} className='go-back-button'>
        <img src={goBackIcon} alt='Go Home Page' className='go-back-icon'/>
      </button>
    )
}

export default GoBack;