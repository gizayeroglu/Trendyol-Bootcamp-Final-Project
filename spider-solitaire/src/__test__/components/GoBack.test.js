import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount, shallow } from 'enzyme';
// import { render, fireEvent } from '@testing-library/react';
// import { MemoryRouter } from 'react-router-dom';

import GoBack from '../../components/GoBack/GoBack';

configure({adapter: new Adapter()});

describe('GoBack component test', function () {

  let wrapper;

  beforeEach(() => { 
    wrapper = shallow(<GoBack />);
  })

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should check backbutton class', () => {
    const buttonTextCheck = wrapper.find('button').hasClass('go-back-button')
    expect(buttonTextCheck).toBeTruthy();
  })
  
//   describe('RouteNotFound', () => {
//   it('Redirects to correct URL on click', () => {
//     const mockHistoryPush = jest.fn();

//   jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useHistory: () => ({
//     push: jest.fn()
//   })
// }));
//     const button = wrapper.find('.go-back-button');
//     button.simulate('click');
//     expect(mockHistoryPush).toHaveBeenCalledWith('/');
//   });
// });
});

