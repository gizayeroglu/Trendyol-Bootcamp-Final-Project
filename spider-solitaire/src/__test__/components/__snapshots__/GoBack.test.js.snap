// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBack component test should render 1`] = `ShallowWrapper {}`;
